File: Airline_Reservation_Backend_Springboot/src/main/java/com/edubridge/service/UserService.java
package com.edubridge.service;

import java.util.List;

import com.edubridge.entity.User;


public interface UserService {
    
public User create(User user);
List<User> getAllUsers();

public User updateUser(User updatedUser);
public  List<User> deleteUser(int id);
public User getUserById(Integer id);

public String deleteUserById(Integer id);
public User loginUser(User user);
public String updateUserById(Integer id);
public List<User> getUsersByRole(String role);
public User findbyusername(String username, String password);
}

File: Airline_Reservation_FrontEnd_Angular/src/app/services/user.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from '../commonurl';
import { User } from '../login/user';
@Injectable({
  providedIn: 'root'
})
export class UserService {
  authenticate(_username: string, _password: string) {
    throw new Error("Method not implemented.");
  }
  updateUserById(_user: User, _id: number) {
    throw new Error('Method not implemented.');
  }

  constructor(private httpClient: HttpClient) { }

  public create(user:User)
  {
  return this.httpClient.post(`${baseUrl}/user/register`, user);
}
public loginUser(user:User)
{
  console.log("user service"+user);
  return this.httpClient.post(`${baseUrl}/user/logincheck`, user);

}
 
public getUsers()
{
  return this.httpClient.get(`${baseUrl}/user/getusers`);
}
 public getUserById(id:number)
 {
return this.httpClient.get(`${baseUrl}/user/byid/${id}`);
 }

public updateUser(user:User,id:number)
{
  return this.httpClient.put(`${baseUrl}/user/update/${id}`, user)
}
public deleteUser(id:number)
{
  return this.httpClient.delete(`${baseUrl}/user/delete/${id}`);
}
public getAllPassengers()
{
return this.httpClient.get(`${baseUrl}/user/get/passenger`);
}


}